README.md

```diff
@@ -10,18 +10,20 @@ This README will provide directions for building, testing, and debugging that co
 
 * NSM_NAME - A string value of network service client name (default "nsc")
 * NSM_CONNECT_TO - A Network service Manager connectTo URL (default "unix:///var/lib/networkservicemesh/nsm.io.sock")
-* NSM_MAX_TOKEN_LIFETIME -  A token lifetime duration (default 24h)                                                            
-* NSM_ROUTES - A comma separated list of routes.                                                                                 
+* NSM_MAX_TOKEN_LIFETIME -  A token lifetime duration (default 24h)
+* NSM_ROUTES - A comma separated list of routes.
 * NSM_LABELS - A list of client labels with format key1=val1,key2=val2, will be used a primary list for network services
-* NSM_MECHANISM - Default Mechanism to use, supported values "kernel","memif"                                                         
+* NSM_MECHANISM - Default Mechanism to use, supported values "kernel", "memif", "vfio"
 * NSM_NETWORK_SERVICES - A list of Network Service Requests URLs with inner format 
-    - [{mechanism}]?:${nsName}[@domainName]?/${interfaceName/memIfSocketName}?${label1}=${value1}&${label2}=${value2}
-        - mechanism one of "kernel", "memif"
+    - \[kernel://]nsName\[@domainName]\[:capability]/interfaceName?\[{labelN=valueN}]
```

Comment:
Do you intend for interfaceName here to be mandatory?

Reply:
Re capabilities, a more careful reading of RFC 2396 indicates we should be OK :)

Turns out properly its:

scheme:[//authority]/path?query

Where authority can be a hostport or can be something else :)

Reply:
Since we are using URI as:
```
[scheme://]userinfo[@host[:port]][/path][?query]
```
Guess it would be better to define **capability** as a part of **userinfo** so it would be:
```
kernel://nsName[:capability][@domainName]/interfaceName[?parameters]
vfio://nsName:capability[@domainName][?parameters]
```

Reply:
Honestly, I think your previous choice of:

vfio://nsName[@domainName][:capability][?parameters]

Is probably better.

Reply:
Moved capability to labels :)